modules/evm/src/lib.rs

```diff
@@ -1363,18 +1368,42 @@ impl<T: Config> Pallet<T> {
 				}
 			});
 
-			ContractStorageSizes::<T>::take(contract);
+			let total_size = ContractStorageSizes::<T>::take(contract);
+			// If there are less than 50 items, we will remove it directly
+			maybe_weight_limit = if total_size
```

Comment:
I think we can always dispatch the task with a hardocded weight limit regardless if it can purge the whole contract or not

Reply:
In this way, if it cannot be completely removed, part of it will be removed, and the rest will be scheduled to remove.